### PR TITLE
Fix content negotiation #120

### PR DESCRIPTION
**Pull request by Karql**
https://github.com/SteeltoeOSS/steeltoe/issues/120#issuecomment-587351045

### CODE REVIEW

File: src/Management/src/EndpointBase/ActuatorMediaTypes.cs [hunk 1]
```diff
@@ -45,6 +45,11 @@ public static string GetContentHeaders(List<string> acceptHeaders, MediaTypeVers
                         contentHeader = AllowedAcceptHeaders(version)
                             .FirstOrDefault(header => acceptHeader == header);
                     }
+
```
Comment (hananiel) [inline src/Management/src/EndpointBase/ActuatorMediaTypes.cs]: This is doing the correct thing if the highest version comes first in the list from client. I am not sure we can depend on that necessarily. Can you rework it to find the highest matching version. Eventually, we need to make the change to setup both versions and respond with the correct version of the response based on header. So the result needs to be(for now):  
```
                    // Configured version  | Accept        | Result
                    //          V1         |  ANY          |   V1   
                    //                     |  No Match     |   APPJSON
                    //                     |  V1           |   V1
                    //                     |  V2           |   V1
                    //                     |  V1,V2/V2/V1  |   V1
                    //          V2         |  ANY          |   V2
                    //                     |  No Match     |   APPJSON
                    //                     |  V1           |   APPJSON
                    //                     |  V2           |   V2
                    //                     |  V1,V2/V2,V1  |   V2

```
What you have does this, it wont work if it comes in a different order.

Comment (Karql) [reply src/Management/src/EndpointBase/ActuatorMediaTypes.cs]: Done ;)

I've also refactored tests for it.

File: src/Management/src/EndpointCore/ContentNegotiation/ContentNegotiationExtensions.cs [hunk 2]
```diff
@@ -48,24 +49,14 @@ public static void LogContentType(this ILogger logger, IHeaderDictionary request
...
+            var acceptMediaTypes = headers.Accept?.Select(x => x.MediaType.Value).ToList();
 
             var contentType = ActuatorMediaTypes.GetContentHeaders(acceptMediaTypes, version);
 
             responseHeaders.Add("Content-Type", contentType);
 
             logger?.LogContentType(requestHeaders, contentType);
         }
-
-        public static bool HasSupportedAcceptHeaders(this IHeaderDictionary requestHeaders, MediaTypeVersion version = MediaTypeVersion.V2)
```
Comment (Karql) [inline src/Management/src/EndpointCore/ContentNegotiation/ContentNegotiationExtensions.cs]: It was not used anywhere

Comment (hananiel) [reply src/Management/src/EndpointCore/ContentNegotiation/ContentNegotiationExtensions.cs]: We can drop that one.

**Pull request merged**